test/validate/pgv_test.cc

```diff
@@ -1,33 +1,70 @@
 #include <iostream>
 #include <cstdlib>
 
-#include "test/validate/test.pb.validate.h"
+// We don't use all the headers in the test below, but including them anyway as
+// a cheap way to get some C++ compiler sanity checking.
+#include "api/bootstrap.pb.validate.h"
+#include "api/protocol.pb.validate.h"
+#include "api/cds.pb.validate.h"
+#include "api/eds.pb.validate.h"
+#include "api/lds.pb.validate.h"
+#include "api/rds.pb.validate.h"
+#include "api/rds.pb.validate.h"
+#include "api/filter/accesslog/accesslog.pb.validate.h"
+#include "api/filter/http/buffer.pb.validate.h"
+#include "api/filter/http/fault.pb.validate.h"
+#include "api/filter/http/health_check.pb.validate.h"
+#include "api/filter/http/lua.pb.validate.h"
+#include "api/filter/http/router.pb.validate.h"
+#include "api/filter/http/transcoder.pb.validate.h"
+#include "api/filter/network/http_connection_manager.pb.validate.h"
+#include "api/filter/network/mongo_proxy.pb.validate.h"
+#include "api/filter/network/redis_proxy.pb.validate.h"
+#include "api/filter/network/tcp_proxy.pb.validate.h"
 
-// Basic protoc-gen-validate C++ validation header inclusion and Validate calls
-// from data-plane-api.
-// TODO(htuch): Switch to using real data-plane-api protos once we can support
-// the required field types.
-int main(int argc, char* argv[]) {
-  {
-    test::validate::Foo empty;
+#include "google/protobuf/text_format.h"
 
+template <class Proto> struct TestCase {
```

Comment:
Separating this into two separate classes might be useful, but we should probably do that as part of a more extensive testing framework for the validation annotations.